src/main/java/io/jsonwebtoken/impl/DefaultJwtBuilder.java

```diff
@@ -294,25 +301,19 @@ public String compact() {
 
         String base64UrlEncodedHeader = base64UrlEncode(jwsHeader, "Unable to serialize header to json.");
 
-        String base64UrlEncodedBody;
+        byte[] bytes;
+        try {
+            bytes = this.payload != null ? payload.getBytes(Strings.UTF_8) : toJson(claims);
```

Comment:
Should you also trim the `this.payload` and check if it is not empty before getting the bytes?

Reply:
Not really?  I mean, leading and trailing space could be a valid payload, and we don't want to sanitize people's payloads.  That said, this code isn't new - it's the same as it was before, just moved up a couple of lines, so I wouldn't want to change it unless there was a strong reason (e.g. in the RFC) to do so.  Or at the very least open a new issue to track it and only do it on a 1.0.0 release.

Reply:
Alright, just wanted to confirm 👍